packages/frontend/src/Build.svelte

```diff
@@ -0,0 +1,370 @@
+let buildImageName: string;
+let buildTag: string;
+let buildEngineId: string;
+
+// Build options
+let buildFolder: string;
+let buildType: string;
+let buildArch: string;
+
+// Other variables
+let success = false;
+let buildInProgress = false;
+let bootcAvailableImages: any[] = [];
+let errorMessage = '';
+let errorFormValidation = '';
+
+// Function that will use listHistoryInfo, if there is anything in the list, pick the first one in the list (as it's the most recent)
+// and fill buildFolder, buildType and buildArch with the values from the selected image.
+async function fillBuildOptions() {
+  const historyInfo = await bootcClient.listHistoryInfo();
+  if (historyInfo.length > 0) {
+    const latestBuild = historyInfo[0];
+    buildFolder = latestBuild.folder;
+    buildType = latestBuild.type;
+    buildArch = latestBuild.arch;
+
+    // Go through bootcAvailableImages and find the one that matches latestBuild.image and latestBuild.tag
+    // and set selectedImage to that value.
+    for (const image of bootcAvailableImages) {
+      if (image.RepoTags && image.RepoTags.length > 0) {
+        if (image.RepoTags[0] === `${latestBuild.name}:${latestBuild.tag}`) {
+          selectedImage = image;
+          break;
+        }
+      }
+    }
+  }
+}
+
+async function buildBootcImage() {
+  // Before ANYTHING validate / check all values before submitting:
+  // Reset the form validation error message
+  errorFormValidation = '';
+  let missingFields = [];
+  if (!buildImageName) missingFields.push('image name');
+  if (!buildTag) missingFields.push('tag');
+  if (!buildEngineId) missingFields.push('engine ID');
+  if (!buildFolder) missingFields.push('build folder');
+  if (!buildType) missingFields.push('build type');
+  if (!buildArch) missingFields.push('architecture');
+  if (missingFields.length > 0) {
+    errorFormValidation = `Missing fields: ${missingFields.join(', ')}`;
+    return; // Stop execution if any field is missing
+  }
+
+  // The build options
+  const buildOptions: BootcBuildInfo = {
+    name: buildImageName,
+    tag: buildTag,
+    engineId: buildEngineId,
+    folder: buildFolder,
+    type: buildType,
+    arch: buildArch,
+  };
+  buildInProgress = true;
+  try {
+    // Do not await.. just start the build.
+    // the reason being is that the validation / error logic happens in buildDiskImage
+    // in the backend and it will error out there as that is where we can console.log
+    // as well as notify the user of the error via showErrorMessage / showInformationMessage, etc.
+    bootcClient.buildImage(buildOptions);
+
+    // Continue doing listHistoryInfo until the build container name, tag, type and arch show up
+    // this means we can safely exit and see it in the dashboard as it's now in the history / running in the background.
+    let timeout = 0;
+    const timeoutLimit = 15; // Timeout after 15 seconds.. this is the "max" amount of time for the API anyways... so that too would error out / console.
```

Comment:
timeout is from which part ?

Reply:
https://github.com/cdrage/podman-desktop-extension-bootc/blob/9f57c8e45fd06762a5f3f96686db9a1e12e733b4/packages/shared/src/messages/MessageProxy.ts#L204-L205 which is 10 seconds. I've updated the comment on the part of the code. 

I purposely set a timeout larger than the API, as the API through MessageProxy should reply back immediately. This is how aistudio is doing it too.

I'll update the PR. I had an incorrect comment on there.